### ISSUE TITLE
MacOS Accessibility: Auto complete causes focus to be completely lost

### ISSUE THREAD

**Issue by Ghorthalon** [microsoft/vscode#246815]
Type: <b>Bug</b>

On latest MacOS and VoiceOver, typing inside a code editor and triggering an auto complete will cause focus to be lost. From then on, arrowing inside the auto complete view is not read, and dismissing the auto complete view with escape or tab doesn't move focus back to the editor. This happens in VS Code 1.99.3 and also latest insiders. 

Steps to reproduce:

1. Enable VoiceOver. Make sure Cursor Tracking is enabled (vo+shift+f3 by default until it says Cursor Tracking on, which should be the default state).
2. Open any file in VS Code and start typing in a way which causes an auto complete to trigger. For example, opening a TypeScript file and typing "export" or "public".
3. Observe focus gets lost and we get stuck somewhere inside VS Code. We can't type or select anything in the auto complete view, if it is even still open. 

Expected (and previous) behavior:
1. We start typing inside a code editor which triggers auto complete
2. Auto complete results are read, we can arrow around to select one, we can insert or dismiss auto complete and/or we can continue typing.
3. Focus gets reliably put back inside the code editor where it was.

VS Code version: Code - Insiders 1.100.0-insider (Universal) (f56819db693dba84feb3716e278b5be0a7378633, 2025-04-17T05:10:42.956Z)
OS version: Darwin arm64 24.4.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M2 Max (12 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 3, 8|
|Memory (System)|32.00GB (0.22GB free)|
|Process Argv|--crash-reporter-id 1823b924-8345-4dac-b16c-f47ec49e2065|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
csharp|ms-|2.72.34
vscode-dotnet-runtime|ms-|2.3.3
indent-report|ogo|1.0.0
lua|sum|3.14.0
hammerspoon|vir|0.5.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt551cf:31249598
vscod805:30301674
vscaac:30438845
c4g48928:30535728
vscrp:30624060
2i9eh265:30646982
962ge761:30841072
dsvsc014:30777825
dsvsc015:30821418
2e7ec940:31000449
pythontbext0:30879054
cppperfnew:30980852
dwnewjupyter:31046869
pythonrstrctxt:31093868
nativeloc1:31118317
e80f6927:31120813
dwcopilot:31158714
5b1c1929:31184661
6074i472:31201624
dwoutputs:31242946
customenabled:31248079
hdaa2157:31222309
copilot_t_ci:31222730
g012b348:31231168
cp15370_t:31280197
pythoneinst12:31251391
treesitterini:31278752
bgtreat:31252123
c7cif404:31283753
31787653:31262186
3e8i5726:31271747
pythonpulldiag:31287486
6gi0g917:31259950
61ibe868:31282443
996jf627:31264550
pythonrdcb7:31268811
usemplatestapi:31288272
gh9da704:31280609
747dc170:31275146
g20af354:31278749
6518g693:31287811
pylancecolorcf:31281600
generatesymbolt:31280541
convertfstringf:31280702
gendocf:31280703
3g1ci329:31285109

```

</details>
